### PR TITLE
Support rails 5.2

### PR DESCRIPTION
**Pull request by islam-taha**
In rails 5.2 string callbacks were removed, so in this PR i've converted conditional strings into procs instead.

Also I've configured travis correspondingly to support only rails 5 in version 0.3.0.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
 
 rvm:
-  - 2.0.0
-  - 2.1.6
-  - 2.2.3
+  - 2.4.5
+  - 2.5.3
 
 gemfile:
-  - gemfiles/ar40.gemfile
```
Comment (mnelson) [inline .travis.yml]: I'd like to keep activerecord 4.2 compatibility as Guideline is still using that (upgrade to rails 5 is happening this quarter)

Comment (islam-taha) [reply .travis.yml]: @mnelson I've changed this back and added activerecord 4.x compatibility.

Comment (mnelson) [reply .travis.yml]: Thanks. Looks like the test suite is very red at the moment.

Comment (islam-taha) [reply .travis.yml]: sorry for this, it's fixed now.

**Comment by mnelson** [comment]
@islam-taha thanks for the help. Version 0.3.0 has been published: https://rubygems.org/gems/stator/versions/0.3.0

**Pull request merged**